shell/platform/fuchsia/flutter/session_connection.cc

```diff
@@ -29,6 +29,25 @@ SessionConnection::SessionConnection(
   session_wrapper_.set_error_handler(
       [callback = session_error_callback](zx_status_t status) { callback(); });
 
+  // Set the |fuchsia::ui::scenic::OnFramePresented()| event handler that will
+  // fire every time a set of one or more frames is presented.
+  session_wrapper_.set_on_frame_presented_handler(
+      [this, handle = vsync_event_handle_](
+          fuchsia::scenic::scheduling::FramePresentedInfo info) {
+        size_t num_presents_handled = info.presentation_infos.size();
+
+        // Since this event is fired on a per-presentation not per-submission
+        // basis, we may have had multiple Present2() calls all end here.
```

Comment:
This is a bit confusing. How about something along the the lines:
// Frame presented: update presents in flight with the remaining unhandled present requests.